nesh/02_customize_your_cluster.ipynb

```diff
@@ -0,0 +1,364 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# How to customize your Dask jobqueue cluster\n",
+    "\n",
+    "covers the following aspects, i.e. how to\n",
+    "* choose a NEC Linux cluster queue\n",
+    "* adjust Dask jobqueue worker resources\n",
+    "* scale Dask clusters adaptively"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Choose a NEC Linux cluster queue\n",
+    "\n",
+    "The [NEC Linux cluster system](https://www.rz.uni-kiel.de/de/angebote/hiperf/nec-linux-cluster) has a theoretical number of 198 batch nodes with 24-32 CPUs and 128-384 GB memory available per execution host. Dask jobqueue workers are intended to have rather short lifetimes and would therefore survive with the walltime limits set for the `clexpress` batch queue. However, the very limited node/execution host number might lead to very unpredictable Dask worker job execution starting times (note the [spiky workload tendency](https://github.com/ExaESM-WP4/nesh-monitoring/blob/master/analysis.ipynb)), which is not good for a default configuration. Therefore, the batch class `clmedium` has been chosen as default. A [relatively stable number of 200-400 idle CPUs](https://github.com/ExaESM-WP4/nesh-monitoring/blob/master/analysis.ipynb) (at least during the considered log period!) is observed and (not too big!) Dask jobqueue clusters should always immediately start/connect. However, as available resources generally fluctuate with present user behaviour (which might change!), below you find an example on how to get an overview on currently available CPU resources and how to set up your Dask jobqueue cluster in a different than the default batch class."
```

Comment:
Regarding the permalinks: You should also use https://nbviewer.jupyter.org/ links, because Github is very unrealiable in directly displaying notebooks.

Example for the spiky workload link: <https://nbviewer.jupyter.org/github/ExaESM-WP4/nesh-monitoring/blob/b6bcdd1a9f522751769de0c3526ab4740b1ef6aa/analysis.ipynb>